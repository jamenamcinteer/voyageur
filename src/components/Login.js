import React from "react";
import styled from "styled-components";
import {} from "twix";
import ButtonA from "./Buttons/ButtonA";

const App = styled.div`
  height: 100vh;
  width: 100vw;
  padding: 40px;
  display: flex;
  align-items: center;
  text-align: center;
  position: relative;

  &:before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: url("/login-background.jpg");
    background-size: cover;
    background-position: top left;
    filter: blur(1px) brightness(90%) grayscale(50%) sepia(10%);
  }
`;

const AppContainer = styled.div`
  width: 100%;
  position: relative;
`;

const AppTitle = styled.h1`
  color: #fff;
`;

const AppSubtitle = styled.h2`
  color: #fff;
`;

const Login = props => {
  return (
    <div>
      <App>
        <AppContainer>
          <AppTitle>Voyageur</AppTitle>
          <AppSubtitle>Travel Budget Planner</AppSubtitle>
          <ButtonA
            to="/auth/google"
            buttonText="Sign In with Google"
            buttonType="primary"
            customStyles={{ background: { display: "inline-block" } }}
          />
          <ButtonA
            to="/auth/facebook"
            buttonText="Sign In with Facebook"
            buttonType="primary"
            customStyles={{ background: { display: "inline-block" } }}
          />
        </AppContainer>
      </App>
    </div>
  );
};

export default Login;
