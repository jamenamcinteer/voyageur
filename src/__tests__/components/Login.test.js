import React from "react";
import { render, cleanup } from "@testing-library/react";
import Login from "../../components/Login";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Login />);
  expect(asFragment()).toMatchSnapshot();
});
