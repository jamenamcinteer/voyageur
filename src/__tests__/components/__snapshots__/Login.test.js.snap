// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat eTyPZp"
  >
    <div
      class="sc-bxivhb fACChQ"
    >
      <h1
        class="sc-ifAKCX kxMoYE"
      >
        Voyageur
      </h1>
      <h2
        class="sc-EHOje iMeQMt"
      >
        Travel Planner
      </h2>
      <a
        class="sc-bdVaJa gtQkNQ"
        href="/auth/google"
        style="display: inline-block;"
      >
        <span
          class="sc-bwzfXH hAnPLf"
          color="#fff"
        >
          Sign In with Google
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
